src/components/survey/other-structure.tsx

```diff
@@ -0,0 +1,308 @@
+import { Stack, Button, Typography, TextField, Accordion, AccordionSummary, AccordionDetails, Box, List } from "@mui/material";
+import Api from "api";
+import { useAppSelector } from "app/hooks";
+import { ErrorContext } from "components/error-handler/error-handler";
+import { selectKeycloak } from "features/auth-slice";
+import { Building, OtherStructure } from "generated/client";
+import strings from "localization/strings";
+import * as React from "react";
+import GenericDialog from "components/generic/generic-dialog";
+import WithDebounce from "components/generic/with-debounce";
+import { Add, ExpandMore } from "@mui/icons-material";
+
+/**
+ * Component properties
+ */
+interface Props {
+  surveyId: string;
+}
+
+/**
+ * Component for other structures
+ */
+const OtherStructures: React.FC<Props> = ({ surveyId }) => {
+  const keycloak = useAppSelector(selectKeycloak);
+  const errorContext = React.useContext(ErrorContext);
+  const [ building, setBuilding ] = React.useState<Building>();
+  const [ creatingOtherStructure, setCreatingOtherStructure ] = React.useState(false);
+  const [ deletingOtherStructureId, setDeletingOtherStructureId ] = React.useState<number>();
+  const [ newOtherStructureName, setNewOtherStructureName ] = React.useState("");
+  const [ newOtherStructureDesc, setNewOtherStructureDesc ] = React.useState("");
```

Comment:
Combine these two into single state that contains the new OtherStructure object.